android/localbuild.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+usage() {
+  echo "usage: $0 [-p platform][--platform platform] [-f flavor][--flavor flavor]"
+  echo "-p platform, --platform platform:         the platform to build for (platform options: x64 (default), x86, arm, arm64)"
+  echo "-f flavor, --flavor flavor:               the build flavor (flavor options: debug (default), release)"
+  echo "-h, --help:                               this help message"
+}
+
+# defining params
+
+BUILD_PLATFORM="x64" # default
+BUILD_FLAVOR="debug"
+SOURCE_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+SOURCES_PATH="$SOURCE_DIR" # this won't be supplied by the user; should be android folder
+OUTPUT_PATH="$SOURCE_DIR/out"
+
+while [ "${1:-}" != "" ]; do
```

Comment:
>${1:-} [](http://example.com/codeflow?start=9&length=6)

(nit) These shell languages are hardly readable. Can we convert this to PowerShell, JavaScript, Perl, or any other non-shell language?

Reply:
Unless this project is meant to run only on Linux hosts, converting the build logic to PowerShell (which also runs on Linux) gives us a more uniform code base.

In my opinion, bash-only complex logic should only be written for UNIX-only projects.